src/zkp/pifac.rs

```diff
@@ -454,13 +451,9 @@ mod tests {
                 &mut rng2,
             )?;
 
-            assert!(small_proof
-                .verify(small_input, &(), &mut transcript())
-                .is_err());
+            assert!(small_proof.verify(input, &(), &mut transcript()).is_err());
```

Comment:
Now this is testing the same thing as the previous test (`incorrect_factors`): the (small) secret inputs are not the factors of the modulus in `input`.

Do you think there's value in testing inputs =/= modulus in two ways? If so, you can keep it but the documentation on L443 needs to change to describe the two ways in which this test fails (the factors do not correspond to the modulus and the factors are small).